Mathlib/GroupTheory/GroupAction/Prod.lean

```diff
@@ -209,3 +209,101 @@ def smulMonoidHom [Monoid α] [MulOneClass β] [MulAction α β] [IsScalarTower
 #align smul_monoid_hom_apply smulMonoidHom_apply
 
 end BundledSMul
+
+section Action_by_Prod
+
+variable (M N α) [Monoid M] [Monoid N]
+
+@[to_additive AddAction.prodEquiv.toFun]
+private abbrev MulAction.prodEquiv.toFun [MulAction (M × N) α] :
+    Σ' (_ : MulAction M α) (_ : MulAction N α), SMulCommClass M N α :=
```

Comment:
You can use `MulAction.compHom α (.inl M N)` here. Why do you use `private abbrev`s instead of normal `def`s?

Reply:
Great! This is now inlined.
I used `private` partly to avoid adding docstrings, and partly because it's an implementation detail of the Equiv. I now realize it's not appropriate here as the user may want to access the definition.
I used `abbrev` because it may be used to construct instances, see [library note](https://leanprover-community.github.io/mathlib_docs/notes.html#reducible%20non-instances).